dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/clouddumper/MetadataRetrieverHttpClientImpl.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.clouddumper;
+
+import java.io.IOException;
+import java.util.Optional;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.core5.http.ClassicHttpRequest;
+import org.apache.hc.core5.http.HttpException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.support.ClassicRequestBuilder;
+import org.apache.http.HttpStatus;
+
+public class MetadataRetrieverHttpClientImpl implements MetadataRetriever {
```

Comment:
Nit: Historically we would have called this HttpClientMetadataRetriever.

Reply:
Sure. I renamed it. I like it the other way around because sorting the classes alphabetically will group together interface and implementations.